sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TablesJacksonSerializer.java

```diff
@@ -104,7 +104,11 @@ public String serialize(Object object, SerializerEncoding encoding) throws IOExc
 
     @Override
     public <U> U deserialize(byte[] bytes, Type type, SerializerEncoding encoding) throws IOException {
-        return deserialize(new ByteArrayInputStream(bytes), type, encoding);
+        if (bytes == null || bytes.length == 0) {
```

Comment:
This change looks good to me to resolve the issue you mentioned. Longer term I'm wondering if this class should move away from sub-typing JacksonAdapter to a composition style pattern. Another solution would be implementing custom JsonSerializer, JsonDeserializer classes which handle the few edge cases this class appears to have been created to handle.